src/LayerPanel/LayerPanelLayersPage/index.js

```diff
@@ -297,10 +310,12 @@ class LayerPanelLayersPage extends Component {
     this.handleMasterCheckbox()
   }
 
-  handleFeatureCheckbox = (layer, feature) => {
+  handleFeatureCheckbox = memo((layer, feature) => {
     feature.set('_ol_kit_feature_visibility', !feature.get('_ol_kit_feature_visibility'))
     this.handleLayerCheckbox(layer)
-  }
+  })
+
+  handleFeatureDoubleClick = memo(this.props.handleFeatureDoubleClick)
```

Comment:
the `memo` wrappers here make sure that every feature item that is rendered have callbacks pointing to the same instance of these methods (instead of instantiating a new function for each feature)